### PR TITLE
Create composite action to cache maven.

### LINKED ISSUE #21423: CI: create composite action to set up Maven cache
## Description

Part of #18750. Create a composite GHA that can be reused from other GHA workflows which sets up Maven (wrapper) to use GHA cache according to [our caching strategy](https://github.com/camunda/camunda/wiki/CI-&-Automation#caching-strategy). That means the action should always try to restore from cache but only save the cache `on: push` to `main` or `stable*` branch builds.

Existing logic (relevant parts) from Zeebe team in https://github.com/camunda/camunda/blob/fd2e86a59785408bd39daa1134aa1d292281c390/.github/actions/setup-zeebe/action.yml#L113-L180 can be reused and moved to a new composite GHA stored in `.github/action/setup-maven-cache`.

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Create composite action to cache maven. wraps the logic in setup-zeebe action in a composite action.

Usage of this composite action requires as inputs the maven cache key modifier.

```yaml
    - name: Configure Maven
      uses: ./.github/actions/setup-maven-cache
      with:
        maven-cache-key-modifier: ${{ inputs.maven-cache-key-modifier }}
```

To test the viability of the composite it is now used in setup-zeebe action.

## Checklist

- [x] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [x] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes https://github.com/camunda/camunda/issues/21423

### CODE REVIEW

File: .github/actions/setup-maven-cache/action.yaml
Comment (cmur2) [inline .github/actions/setup-maven-cache/action.yaml:11]: This input is unneeded and should be removed. All steps in this action are conditional on this input so it is easier to just not call the action if the caller identifies that no Java setup is desired.

File: .github/actions/setup-zeebe/action.yml
Comment (cmur2) [inline .github/actions/setup-zeebe/action.yml:116]: Convert this to an `if: {{ inputs.java == 'true' }}` conditional for the step that calls the setup-maven-cache action.

**Comment by cmur2** [comment]
@rodrigo-lourenco-lopes since you asked how to test this PR:

I looked at the CI runs and can see e.g. in https://github.com/camunda/camunda/actions/runs/10598267866/job/29370652128?pr=21685 that the new composite action is used as expected and restoring a cache from the `main` branch successfully. That means to me that the cache restoring part works fine and needs no further testing.

It should also be tested if a cache can be created successfully (to avoid roundtrip of merging this PR to `main` and then realizing it doesn't work). This can be done by temporarily tweaking/inverting the `if` conditions in the new composite action so that they allow creating a cache on a PR.

**Comment by rodrigo-lourenco-lopes** [comment]
> It should also be tested if a cache can be created successfully (to avoid roundtrip of merging this PR to `main` and then realizing it doesn't work). This can be done by temporarily tweaking/inverting the `if` conditions in the new composite action so that they allow creating a cache on a PR.

Thanks for the tip! 🙌 I will add a commit to test it in this PR while it is in draft, and revert it later once it is ready to merge and if the test is successful.

**Comment by rodrigo-lourenco-lopes** [comment]
It seems that the test was is working by observing [here](https://github.com/camunda/camunda/actions/runs/10614567736/job/29442431525?pr=21685). The cache operation was run creating one with the key `github-hosted-macOS-mvn-shared-b8f0f396e1172a55c103b6acfb8bc8217f2fef46e76fe9156890ec1d0b4e4745
` and restoring from the same key right after successfully.

**Comment by cmur2** [comment]
> It seems that the test was is working by observing [here](https://github.com/camunda/camunda/actions/runs/10614567736/job/29442431525?pr=21685). The cache operation was run creating one with the key `github-hosted-macOS-mvn-shared-b8f0f396e1172a55c103b6acfb8bc8217f2fef46e76fe9156890ec1d0b4e4745 ` and restoring from the same key right after successfully.

I don't think that is what happened. In general in the beginning of a job only restores happen, creating a cache is happening in a "Post Run" step. See [this line](https://github.com/camunda/camunda/actions/runs/10614567736/job/29442431525?pr=21685#step:3:183) which restores the cache and then [this line](https://github.com/camunda/camunda/actions/runs/10614567736/job/29442431525?pr=21685#step:3:216) which restores it again.

Creating/saving the cache would happen in [this line](https://github.com/camunda/camunda/actions/runs/10614567736/job/29442431525?pr=21685#step:15:3) but doesn't since a cache was already found (and could be restored).

To test that effectively you probably have to change the cache key temporarily to something unique so no cache restore is possible -> means that cache creation will be triggered at the end.

**Comment by cmur2** [comment]
> To test that effectively you probably have to change the cache key temporarily to something unique so no cache restore is possible -> means that cache creation will be triggered at the end.

@rodrigo-lourenco-lopes can you try this please?

**Comment by cmur2** [comment]
> [here](https://github.com/camunda/camunda/actions/runs/10666794638/job/29563145662?pr=21685#step:15:5) we can see that the cache was created successfully in the post run step like you suggested.
> 
> The test commit had the condition inverted to allow the creation of the cache on PRs and different cache key names so that we would have a cache miss.

Thank you very much for testing this scenario 👍🏼

**Review by cmur2 — APPROVED** [review]
LGTM

**Pull request merged**
